### PR TITLE
Adding support for JSON type in Mysql and also seperating the long types so that length validation isn't an issue

### PR DESCRIPTION
**Pull request by joelstobart**
Add JSON mysql type and also set the rv for text to cf_sql_longvarchar
Add the cf_SQL_LONGVARCHAR and CF_SQL_LONGNVARCHAR to return "text" rather than string

### CODE REVIEW

File: wheels/model/adapters/MySQL.cfc [hunk 1]
```diff
@@ -50,9 +50,12 @@ component extends="Base" output=false {
 			case "varbinary":
 				local.rv = "cf_sql_varbinary";
 				break;
-			case "varchar": case "text": case "mediumtext": case "longtext": case "tinytext": case "enum": case "set":
+			case "varchar": case "enum": case "set":
 				local.rv = "cf_sql_varchar";
 				break;
+			case "json": case "text": case "mediumtext": case "longtext": case "tinytext": 
+                loc.rv = "cf_sql_longvarchar";
```
Comment (chrisdpeters) [inline wheels/model/adapters/MySQL.cfc]: For CFWheels 2.0, this should be using the `local` scope, not the now-defunct `var`ed `loc` struct.

**Comment by chrisdpeters** [comment]
@joelstobart Thank you for this. Can you fix the one issue that I pointed out?

File: wheels/model/adapters/MySQL.cfc (hunk 1 above)
Comment (joelstobart) [inline wheels/model/adapters/MySQL.cfc]: Sorry - I'm running a version older than this. So the syntax was differrent...

**Comment by joelstobart** [comment]
FYI:  Test results

Lucee 5.1.0.34

Status: 
 Passed

Duration
00:00:11

Packages
286

Tests
1361

Failures
0

Errors
0

**Comment by perdjurner** [comment]
Thank you.

Note: I moved "tinytext" back to map to "cf_sql_varchar".

**Pull request merged**